pkg/controller/backup_storage_location_controller.go

```diff
@@ -127,7 +219,7 @@ func (r *backupStorageLocationReconciler) Reconcile(ctx context.Context, req ctr
 				err = errors.Wrapf(err, "BackupStorageLocation %q is unavailable", location.Name)
 				unavailableErrors = append(unavailableErrors, err.Error())
 				location.Status.Phase = velerov1api.BackupStorageLocationPhaseUnavailable
-				location.Status.Message = err.Error()
+				location.Status.Message = sanitizeStorageError(err)
```

Comment:
How about we implement the logic on Azure plugin side? It is Azure specific logic